### DISCUSSION TITLE
Correct size/resolution for recipe images?

### DISCUSSION THREAD

**Discussion by KidA90** [TandoorRecipes/recipes#2331] [Q&A]
Can someone point me in the right direction on what size/resolution/properties images should have to be displayed properly in the recipe view? For me, it seems rather random how Tandoor treats images and displays them in the recipes. 

For example, I have one recipe with an image which is 1.200 x 1.200 px and has a resolution of 72 dpi. In Tandoor it gets displayed like this:
![Bildschirmfoto vom 2023-02-24 14-13-49](https://user-images.githubusercontent.com/58784953/221188827-7c411bdf-ed97-45c8-b035-911e26837af0.png)

At the same time, another recipe has an image with 860 x 573 px and a resolution of 72 dpi and gets displayed properly:
![Bildschirmfoto vom 2023-02-24 14-22-18](https://user-images.githubusercontent.com/58784953/221189113-21945cd6-0ed6-4aba-a5e9-ccde86e54e04.png)

**Comment by smilerz** [comment]
Tandoor won't resize pictures unless it exceeds the viewport along 1 dimension.  If that occurs it will scale to fit within the viewport.

Edit:  I've never seen the behavior you are showing (I originally read your dimensions in US conventions, so slightly larger than 1px).  If you open that picture in a new tab does it have the correct proportions?

**Reply by KidA90** [reply]
Thank you for your instant reply!

> I originally read your dimensions in US conventions, so slightly larger than 1px

Sorry, force of habit 😅️ I meant 1200 x 1200 px.

> If you open that picture in a new tab does it have the correct proportions?

Yes, if I open the picture in a new tab, it gets displayed in the original size.

Are the width and the height of the viewport static and if so what dimensions are used?

**Reply by KidA90** [reply]
On the recipe overview, the pictures are scaled correctly as well:
![Bildschirmfoto vom 2023-02-24 14-41-43](https://user-images.githubusercontent.com/58784953/221193254-4b5a91f6-261a-4d9d-a32d-2fe5b70e12e6.png)

**Reply by smilerz** [reply]
> Are the width and the height of the viewport static and if so what dimensions are used?

The layout utilizes bootstrap grids which scale to window size.

Which browser do you use?  Assuming its chrome or edge can you go into developer mode (F12 usually) and view 'Elements' & 'Computed'?

If you select the picture it should show you the computed dimensions.
![image](https://user-images.githubusercontent.com/1808639/221198096-710532ad-355b-4ef0-8798-fe185d94a9b3.png)

**Reply by KidA90** [reply]
I'm using Brave Browser, but it is chromium-based like the others. I did go into the developer mode. As you can see, the image gets scaled down to 14 x 14 px:
![Bildschirmfoto vom 2023-02-24 19-19-32](https://user-images.githubusercontent.com/58784953/221260879-53ece0c2-fba4-48db-b4a4-495fd12eb53c.png)


I tried to upload the image with different sizes (down to 100 x 100 px), but the output is always the same (14 x 14 px), even with a totally different image.

BUT: If I add some more ingredients to the recipes, so the list of ingredients on the left has more height, scaling works better:
![image](https://user-images.githubusercontent.com/58784953/221262776-708adf3a-6ce5-4f06-b700-bc5384d69f1a.png)

Nevertheless, I think the image is supposed to be scaled to the same height as the ingredient list, isn't it?

**Reply by smilerz** [reply]
which version of Tandoor are you running?  I'm not able to duplicate that behavior.

Do you get the same results using Edge or Chrome?

**Reply by KidA90** [reply]
I'm running Tandoor 1.4.7 in Docker (Docker Compose). I get the same behaviour in Chrome as well as in Firefox. Strangely, on my mobil phone (using Brave) it looks fine:
![Screenshot_20230224_200951_Brave](https://user-images.githubusercontent.com/58784953/221270916-0a5da57f-5c92-4318-9fed-c510a1a5542c.jpg)

**Reply by smilerz** [reply]
The height of the picture is calculated, and is supposed to be the height of the Ingredient Card - but seems like something is off on that calculation in some way.

Can you open an issue?

**Reply by KidA90** [reply]
Will do, thank you for your support!
